hail/src/main/scala/is/hail/expr/ir/PruneDeadFields.scala

```diff
@@ -415,7 +415,7 @@ object PruneDeadFields {
         val reqRowsType = TStream(requestedType.rowType)
         val bodyDep = memoizeValueIR(body, reqRowsType, memo)
         val depGlobalType = unifySeq(child.typ.globalType,
-          bodyDep.eval.lookupOption(gName).map(_.result()).getOrElse(Array()))
+          bodyDep.eval.lookupOption(gName).map(_.result()).getOrElse(Array()) :+ requestedType.globalType)
```

Comment:
Also worth looking here. This was a bug I uncovered while doing this. We need to consider the requested type when doing the `PruneDeadFields` pass for `TableMapPartitions`.

Reply:
ahh, good